### PR TITLE
Update default tag in Helm charts for 1.2

### PR DESCRIPTION
**Pull request by rlenglet**
Keep consistent with istio/istio, cf. https://github.com/istio/istio/pull/14813

### CODE REVIEW

**Comment by tiswanso** [comment]
@sdake -- this is a sample override file for gke (`values_gke.yaml`) so we don't need as much caution here, IMO.  Probably, we can just remove those settings and any other copied values from the file.  No tests or demos use the file at this point.

File: deployments/kubernetes/install/helm/istio-cni/values.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 hub: gcr.io/istio-release
-tag: master-latest-daily
+tag: release-1.2-latest-daily
```
Comment (sdake) [inline deployments/kubernetes/install/helm/istio-cni/values.yaml]: @tiswanso this is not a gke override -this is the main config.  The test-infra seds (iirc) master-latest-daily to produce the version number.

Comment (tiswanso) [reply deployments/kubernetes/install/helm/istio-cni/values.yaml]: yeah, my bad... there was a similar fix yesterday on master for the gke file only and I assumed this was a cherrypick.

**Comment by tiswanso** [comment]
@rlenglet -- see this PR from @utka for what @sdake is referring to for CNI:  https://github.com/istio/istio/pull/14806/files

**Comment by tiswanso** [comment]
@sdake @rlenglet -- the sed in istio/istio#14806 should still work with the diffs in this PR.  It matches on `tag: .*-latest-daily`

**Comment by sdake** [comment]
Thanks @tiswanso for hunting that down.  I was looking for it :)

**Pull request merged**